druid/src/widget/label.rs

```diff
@@ -17,25 +17,36 @@
 use crate::piet::{Color, PietText};
 use crate::widget::prelude::*;
 use crate::{
-    BoxConstraints, Data, FontDescriptor, KeyOrValue, LocalizedString, Point, Size, TextAlignment,
-    TextLayout,
+    ArcStr, BoxConstraints, Data, FontDescriptor, KeyOrValue, LocalizedString, Point, Size,
+    TextAlignment, TextLayout,
 };
 
 // added padding between the edges of the widget and the text.
 const LABEL_X_PADDING: f64 = 2.0;
 
 /// The text for the label.
 ///
-/// This can be one of three things; either a `String`, a [`LocalizedString`],
+/// This can be one of three things; either a [`ArcStr`], a [`LocalizedString`],
 /// or a closure with the signature, `Fn(&T, &Env) -> String`, where `T` is
-/// the `Data` at this point in the tree.
+/// the [`Data`] at this point in the tree.
 ///
+/// [`ArcStr`]: ../type.ArcStr.html
 /// [`LocalizedString`]: ../struct.LocalizedString.html
+/// [`Data`]: ../trait.Data.html
 pub enum LabelText<T> {
     /// Localized string that will be resolved through `Env`.
     Localized(LocalizedString<T>),
-    /// Specific text
-    Specific(String),
+    /// Static text.
+    Static {
+        /// The text.
+        string: ArcStr,
+        /// Whether or not the `resolved` method has been called yet.
+        ///
+        /// We want to return `true` from that method when it is first called,
+        /// so that callers will know to retrieve the text. This matches
+        /// the behaviour of the other variants.
+        resolved: bool,
+    },
```

Comment:
Maybe move this into its own `Static` struct the way we did with `Dynamic`? Then we could hide the `resolved` field.